src/acp/rule.test.ts

```diff
@@ -0,0 +1,704 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { jest, describe, it, expect } from "@jest/globals";
+import { asIri, createThing } from "../thing/thing";
+import {
+  addForbiddenRuleToPolicy,
+  addOptionalRuleToPolicy,
+  addRequiredRuleToPolicy,
+  getForbiddenRuleOnPolicyAll,
+  getOptionalRuleOnPolicyAll,
+  getRequiredRuleOnPolicyAll,
+  Rule,
+  setForbiddenRuleOnPolicy,
+  setOptionalRuleOnPolicy,
+  setRequiredRuleOnPolicy,
+} from "./rule";
+import { DataFactory } from "n3";
+import { Thing, ThingPersisted } from "../interfaces";
+import { Policy } from "./policy";
+
+const ACP_ANY = "http://www.w3.org/ns/solid/acp#anyOf";
```

Comment:
Why are these 3 local constants defined here as strings (and not NamedNodes)? Each of them is used 12 times in this file - 11 of those times they're wrapped in a NameNode call, and the 12th time they could be passed as a NamedNode or a string! I'm not trying to kick off the String vs IRI debate again here (that's a totally different discussion anyway) - it just seems a clear example of making this test code **_less_** readable and unnecessarily verbose for not benefit whatsoever.

Reply:
There is not particular reason, so they could be changed to be `NamedNodes`, but they are mostly used along other named nodes, such as: 
```
expect(
      myPolicy.has(
        DataFactory.quad(
          DataFactory.namedNode("https://some.pod/policy-resource#policy"),
          DataFactory.namedNode(ACP_NONE),
          DataFactory.namedNode("https://some.pod/rule-resource#rule")
        )
      )
    ).toBe(true);
```
so there would be little value in changing `DataFactory.namedNode(ACP_NONE)` into `ACP_NONE`. To make the statement more compact, we'd have to also declare constants for `DataFactory.namedNode("https://some.pod/policy-resource#policy")` and `DataFactory.namedNode("https://some.pod/rule-resource#rule")`, which I'm happy to do but I wasn't sure would make the tests more readable. i'll give it a try.

Reply:
The similar code snippet, using named nodes constants, would be change as in the following: 
**Before**
```
const myPolicy = addOptionalRuleToPolicy(
      mockPolicy("https://some.pod/policy-resource#policy"),
      mockRule("https://some.pod/rule-resource#rule")
    );
    expect(
      myPolicy.has(
        DataFactory.quad(
          DataFactory.namedNode("https://some.pod/policy-resource#policy"),
          DataFactory.namedNode(ACP_ANY),
          DataFactory.namedNode("https://some.pod/rule-resource#rule")
        )
      )
    ).toBe(true);
```
**After**
```
it("adds the rule in the forbidden rules of the policy", () => {
    const myPolicy = addForbiddenRuleToPolicy(
      mockPolicy(MOCKED_POLICY_IRI),
      mockRule(MOCKED_RULE_IRI)
    );
    expect(
      myPolicy.has(
        DataFactory.quad( MOCKED_POLICY_IRI, ACP_NONE, MOCKED_RULE_IRI )
      )
    ).toBe(true);
  });
```
It is indeed more compact, but I'm not sure it is more readable. What do you think ?

Reply:
Yeah, I think it's much more readable. Just to add too though, the `MOCKED_POLICY_IRI` etc. values I'd actually move into the existing Inrupt test vocabulary (https://github.com/inrupt/solid-common-vocab-rdf/blob/master/inrupt-rdf/Common/CopyOfVocab/inrupt-test.ttl)  (and as above, I wouldn't refer to them as MOCKs either).

Reply:
Fixed in 5f1e895c4a9284fba126ba97dce97d20e312d00f